src/main/java/net/sf/jabref/logic/formatter/casechanger/Title.java

```diff
@@ -16,6 +17,11 @@ public Title(String title) {
         this.words.addAll(new TitleParser().parse(title));
     }
 
+    public List<Word> getWordsReadOnly() {
```

Comment:
Ah, I get it. The real problem is that the `Word` objects are mutable, less that the list is.

However, that means that you can *still* return `Collections.unmodifiableList(words)` and execute `forEach(Word::toUpperFirst)`. The `unmodifiableList` only ensures that the structure of the list will not change (and your lambda does not change it), the objects contained therein can still be modified (which is ugly, but anyway...). Therefore, I would ask you to implement my suggestion and skip the deprecated getter.